Sources/SwiftGenKit/Parsers/Fonts/CTFont.swift

```diff
@@ -5,27 +5,87 @@
 //
 
 import AppKit.NSFont
+import CoreText
 import Foundation
 import PathKit
 
 extension CTFont {
-  static func parse(file: Path, relativeTo parent: Path? = nil) -> [Fonts.Font] {
+  static func parse(file: Path, relativeTo parent: Path? = nil, extractCodePoints: Bool = false) -> [Fonts.Font] {
     let descs = CTFontManagerCreateFontDescriptorsFromURL(file.url as CFURL) as NSArray?
     guard let descRefs = (descs as? [CTFontDescriptor]) else { return [] }
 
+    // swiftlint:disable:next closure_body_length
     return descRefs.compactMap { desc -> Fonts.Font? in
       let font = CTFontCreateWithFontDescriptorAndOptions(desc, 0.0, nil, [.preventAutoActivation])
       let postScriptName = CTFontCopyPostScriptName(font) as String
       guard let familyName = CTFontCopyAttribute(font, kCTFontFamilyNameAttribute) as? String,
         let style = CTFontCopyAttribute(font, kCTFontStyleNameAttribute) as? String else { return nil }
 
       let relPath = parent.flatMap { file.relative(to: $0) } ?? file
+
+      let characterSet = CTFontCopyCharacterSet(font) as CharacterSet
+      let cgFont = CTFontCopyGraphicsFont(font, nil)
+
+      var icons: [String: String] = [:]
+
+      if extractCodePoints {
+        for unicode in characterSet.codePoints {
+          guard let unicodeScalar = UnicodeScalar(unicode) else { continue }
+
+          let utf16codepoints = Array(Character(unicodeScalar).utf16)
```

Comment:
Wondering if there isn't a more efficient way to go from codepoint to utf16 than CodePoint>UnicodeScalar>Character>.utf16 🤔
Will have to look at the various String/Character/Unicode APIs to double check, as this is often tricky stuff with edge cases so better be sure.

Reply:
Actually there is, and it's more correct: `Array(unicodeScalar.utf16)`